chrome/content/insertSignature.js

```diff
@@ -9,24 +9,27 @@ function startScript () {
     if (previousUrl !== location.href) {
       previousUrl = location.href;
       const hasInlineTooltip = doesInlineTooltipExist();
-      if (hasInlineTooltip) {
+      const hasSignatureBtn = doesSignatureBtnExist();
+      // Check to see if inline-tooltip exists and that there is no previous Signature
+      // button before creating a new one. This corrects the edge case of
+      // `medium.com/new-story` changing to `medium.com/p/<some_hash>/edit` which
+      // resulted in a duplicate Signature button being added.
+      // Refs: https://github.com/cedricium/signature/issues/7
+      if (hasInlineTooltip && !hasSignatureBtn) {
         createSignatureButton();
       }
     }
   }, 250);
 }
 
 function doesInlineTooltipExist () {
-  // edge case: medium.com/new-story --> medium.com/p/<random-hashed-id>/edit
-  // this occurs when the user creates a new story then gives the article some text
-  // to a totally new / blank story.
-  if (location.href === 'https://medium.com/new-story') {
-    return false;
-  }
-
   return document.querySelector('.inlineTooltip');
```

Comment:
Use `!!` when using `document.querySelector` to ensure a boolean is being returned:
```js
 return !!document.querySelector('.inlineTooltip');
```